pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -108,12 +109,17 @@ func NewAttachDetachController(
 		return nil, fmt.Errorf("Could not initialize volume plugins for Attach/Detach Controller: %+v", err)
 	}
 
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	recorder := eventBroadcaster.NewRecorder(api.EventSource{Component: "attachdetach-controller"})
+
```

Comment:
@saad-ali  ok, I will revert back to what I had, sorry for the confusion, and yes, I'm using the same EventBroadcaster that is already created in the run() method, I just moved it up in front of startControllers so I could pass it in.  I removed the initialization of broadcaster from attach_detach_controller.go